src/main/java/com/salesforce/dataloader/action/visitor/BulkApiVisitorUtil.java

```diff
@@ -154,6 +169,10 @@ void createJob(Config cfg) throws AsyncApiException {
                                       "chunkSize=" + this.queryChunkSize + startRowParam);
             }
         }
+        if (isBulkV2QueryJob()) {
+            job.setObject(this.config.getString(Config.EXTRACT_SOQL));
```

Comment:
I am guessing that "EXTRACT_SOQL" is the soql statement. Is that the right config param to assign to "Object"?

Reply:
I agree that this is not a "clean" fix. Unfortunately, JobInfo does not have a field for SoQL because BulkConnection has an explicit method, createBatch, that accepts SoQL as an argument. There in no notion of a batch in Bulk V2. So, the best way to address this side-effect seems to be to extend JobInfo, perhaps as BulkV2JobInfo through inheritance? containment would require a lot of boilerplate code for all public methods of JobInfo.

Reply:
Will await extending JobInfo with a member variable to store the SoQL string and use it instead of setting its Object variable to store the query string to be passed during Bulk V2 query job creation.